rosbag2_transport/test/rosbag2_transport/test_burst.cpp

```diff
@@ -366,3 +399,64 @@ TEST_F(RosBag2PlayTestFixture, burst_bursting_only_filtered_topics) {
   // All we care is that any messages arrived
   EXPECT_THAT(replayed_topic2, SizeIs(Eq(EXPECTED_BURST_COUNT)));
 }
+
+TEST_F(RosBag2PlayTestFixture, burst_bursting_only_filtered_services) {
+  const std::string service_name1 = "/test_service1";
+  const std::string service_name2 = "/test_service2";
+  const std::string service_event_name1 = service_name1 + "/_service_event";
+  const std::string service_event_name2 = service_name2 + "/_service_event";
+
+  auto services_types = std::vector<rosbag2_storage::TopicMetadata>{
+    {1u, service_event_name1, "test_msgs/srv/BasicTypes_Event", "", {}, ""},
+    {2u, service_event_name2, "test_msgs/srv/BasicTypes_Event", "", {}, ""},
+  };
+  std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages =
+  {
+    serialize_test_message(service_event_name1, 500, get_service_event_message_basic_types()[0]),
+    serialize_test_message(service_event_name2, 600, get_service_event_message_basic_types()[0]),
+    serialize_test_message(service_event_name1, 400, get_service_event_message_basic_types()[1]),
+    serialize_test_message(service_event_name2, 500, get_service_event_message_basic_types()[1])
+  };
+
+  std::vector<std::shared_ptr<test_msgs::srv::BasicTypes::Request>> service1_receive_requests;
+  std::vector<std::shared_ptr<test_msgs::srv::BasicTypes::Request>> service2_receive_requests;
+
+  srv_->setup_service<test_msgs::srv::BasicTypes>(service_name1, service1_receive_requests);
+  srv_->setup_service<test_msgs::srv::BasicTypes>(service_name2, service2_receive_requests);
+
+  srv_->run_services();
+
+  // Filter allows test_service2, blocks test_service1
+  play_options_.services_to_filter = {service_event_name2};
+  play_options_.publish_service_requests = true;
+  auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
+  prepared_mock_reader->prepare(messages, services_types);
+  auto reader = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
+  auto player = std::make_shared<MockPlayer>(std::move(reader), storage_options_, play_options_);
+
+  player->pause();
+  ASSERT_TRUE(player->is_paused());
+
+  // Check services are ready
+  ASSERT_TRUE(srv_->all_services_ready());
+
+  rclcpp::executors::SingleThreadedExecutor exec;
+  exec.add_node(player);
+  auto spin_thread = std::thread([&exec]() {exec.spin();});
+  player->play();
+
+  const size_t EXPECTED_BURST_COUNT = 2;
+  ASSERT_EQ(player->burst(EXPECTED_BURST_COUNT), EXPECTED_BURST_COUNT);
+
+  ASSERT_TRUE(player->is_paused());
+  player->resume();
+
+  player->wait_for_playback_to_finish();
+  EXPECT_TRUE(player->wait_for_sent_service_requests_to_finish("", 2s));
```

Comment:
i think we do not need to wait for `service_name1` since it is filtered out?

```suggestion
  EXPECT_TRUE(player->wait_for_sent_service_requests_to_finish(service_name2, 2s));
```

Reply:
If the service name is empty, it will wait for all existing future requests to finish. For service 1, the service client isn't created since filter setting. So current all future requests are sent to service 2.

Reply:
I agree with @Barry-Xu-2018. We need intentionally to wait for all sent services since we have a check underneath that 
`EXPECT_EQ(service1_receive_requests.size(), 0);` otherwise thes test could be false negative.